tests/unit/library/test_json.py

```diff
@@ -1545,7 +1545,8 @@ class TestAdditionalProperties:
         },
     "additionalProperties": {
             "type": "integer"
-        }
+        },
+    "required": ["mystr"]
```

Comment:
Does this addition mean that our previous generations weren't quite correct? If `required` is not _required_ but was always implicitly there, then shouldn't this test have two variants (possibly via an `include_required` pytest parameter) with and with out this? Same for the other test below.

Reply:
I believe (need to check json schema docs) that not including a `required` array implicitly says "nothing is required". Our previous behavior was misaligned with this, instead treating all properties as required no matter what was passed (including passing nothing). This isn't technically *wrong*, as doing so will always produce json that is valid with respect to the schema.

In any case, some of our tests were misspecifications if we actually want to respect the `required` argument, namely the negative tests that assert that some properties exist without passing those properties as required.

I only "fixed" the tests that were failing, and I probably missed a few...

Reply:
We definitely want to explicitly test `required` though (e.g. via parameterizing a `required` array). Haven't gotten there yet.